lib/pgboss.service.ts

```diff
@@ -39,23 +42,32 @@ export class PgBossService {
     data?: TData,
     options?: PgBoss.ScheduleOptions,
   ) {
+    await this.ensureQueueExists(name);
     await this.pgBoss.schedule(name, cron, data ?? {}, options ?? {});
     await this.pgBoss.work<TData>(
       name,
-      { ...options, includeMetadata: true },
+      { ...transformOptions(options), includeMetadata: true },
       handler,
     );
   }
 
   async registerJob<TData extends object>(
     name: string,
     handler: WorkWithMetadataHandler<TData>,
-    options?: PgBoss.BatchWorkOptions,
+    options?: PgBoss.WorkOptions,
   ) {
+    await this.ensureQueueExists(name);
     await this.pgBoss.work<TData>(
       name,
       { ...options, includeMetadata: true },
       handler,
     );
   }
+
+  async ensureQueueExists(queueName: string) {
+    const currentQueue = await this.pgBoss.getQueue(queueName);
+    if (!currentQueue) {
+      await this.pgBoss.createQueue(queueName);
```

Comment:
isnt this automatically creating queue if not exists? do we need to check it again?

Reply:
https://github.com/timgit/pg-boss/issues/467
my bad, came across the issue it's fixed now